### PR TITLE
Upgrade to Node 20

### PR DESCRIPTION
**Pull request by apupier**
VS Code is now based on Node 20
Node 18 is in maintenance mode since September 2023

### CODE REVIEW

**Comment by apupier** [comment]
test in error:

```
  1) Test gitpod Id manager
       Should generate Red Hat UUID from GITPOD_GIT_USER_EMAIL env:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
+ actual - expected

+ 'b45016eb-8171-4cc2-b2ab-6d8920c0508f'
- '465b7cd6-0f77-5fc8-97ed-7b6342df109f'
      + expected - actual

      -b45016eb-8171-4cc2-b2ab-6d8920c0508f
      +465b7cd6-0f77-5fc8-97ed-7b6342df109f
      
      at /home/runner/work/vscode-redhat-telemetry/vscode-redhat-telemetry/src/tests/gitpod/gitpodIdManager.test.ts:27:14
      at step (src/tests/gitpod/gitpodIdManager.test.ts:33:23)
      at Object.next (src/tests/gitpod/gitpodIdManager.test.ts:14:53)
      at fulfilled (src/tests/gitpod/gitpodIdManager.test.ts:5:58)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
 ```

If I understand well, it means that using Node 20 instead of Node 20, the generated id for the users will be modified?

**Comment by fbricon** [comment]
Yeah apparently. Only an issue for the test though

**Comment by fbricon** [comment]
Might be caused by an update to the UUID lib

**Comment by apupier** [comment]
locally with Node 18, the test is passing.
With Node 20.16.0, 20.17.0 (the one used on CI on last builds), they are all generating different uuids.

**Comment by apupier** [comment]
in practice, if I understand well, it only reveals a problem here, not creating a new one.
It would mean that the uuid of the users are changing depending on the version of Node used, which I think in VS Code is embedded so between VS Code versions they might have different uuids

**Comment by apupier** [comment]
the other problem is that every time the Node version will be updated on CI, the tests will start to fail

**Comment by apupier** [comment]
and locally we test will always fail too

**Comment by apupier** [comment]
It is also dependent on the machine it is run onto, the container on CI is changing, so the id is changing between runs.

We got with 2 concurrent run these 2 genrated uuids: 2f780468-f8ea-4cfc-a80f-5d027b8bb4e9 and b45016eb-8171-4cc2-b2ab-6d8920c0508f

**Comment by apupier** [comment]
after investigation, the v4 which means random is used because there is this error:
`Failed to get user id from Gitpod Error: ENOENT: no such file or directory, open '/home/runner/work/vscode-redhat-telemetry/vscode-redhat-telemetry/.redhat/anonymousId'`
So i guess that with Node 20, it is either tryign to create in another place or requires different rights to create it

File: src/tests/gitpod/gitpodIdManager.test.ts [hunk 1]
```diff
@@ -8,23 +8,23 @@ const redhatDir = `${process.cwd()}/.redhat/`;
...
+        fs.rmSync(redhatDir, { recursive: true, force: true });
+      }
     });
     test('Should generate Red Hat UUID from GITPOD_GIT_USER_EMAIL env', async () => {
       env.GITPOD_GIT_USER_EMAIL = 'some.user@company.com';
       console.log(env.GITPOD_GIT_USER_EMAIL);
       const fsIdProvider = new FileSystemIdProvider(redhatDir);
       const gitpod = new GitpodIdProvider(fsIdProvider);
       const id = await gitpod.loadRedHatUUID();
-      const expectedId = '465b7cd6-0f77-5fc8-97ed-7b6342df109f';
```
Comment (fbricon) [inline src/tests/gitpod/gitpodIdManager.test.ts:26]: Actually for a given name + namespace, UUID should be constant, independently of the Node version

See https://www.uuidtools.com/api/generate/v5/namespace/44662bc6-c388-4e0e-a652-53bda6f35923/name/some.user@company.com

**Comment by fbricon** [comment]
also see https://github.com/tschaub/mock-fs/issues/377

File: src/tests/gitpod/gitpodIdManager.test.ts (hunk 1 above)
Comment (apupier) [reply src/tests/gitpod/gitpodIdManager.test.ts]: > Actually for a given name + namespace, UUID should be constant, independently of the Node version
> 
> See https://www.uuidtools.com/api/generate/v5/namespace/44662bc6-c388-4e0e-a652-53bda6f35923/name/some.user@company.com

ah my bad, I set it back to the previous vrsion, i was only mixed up in my head with the various changes.
So the probelm was really just related to the mock-fs which was not working anymore

File: src/tests/gitpod/gitpodIdManager.test.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import * as assert from 'assert';
-import * as mock from 'mock-fs';
```
Comment (fbricon) [inline src/tests/gitpod/gitpodIdManager.test.ts:2]: is mock-fs still used elswhere?  If not package.json+lock should be updated to remove it

Comment (apupier) [reply src/tests/gitpod/gitpodIdManager.test.ts:2]: yes it is, well in a disabled test but required at compilation https://github.com/redhat-developer/vscode-redhat-telemetry/blob/7aa91a44721969edb223b1dee46625a0410de075/src/tests/services/configurationManager.disabled.ts#L4

**Pull request merged**